telegram/message.py

```diff
@@ -338,10 +338,19 @@ def chat_id(self):
 
     @property
     def link(self):
-        """:obj:`str`: Convenience property. If the chat of the message is a supergroup or a
-        channel and has a :attr:`Chat.username`, returns a t.me link of the message."""
-        if self.chat.type in (Chat.SUPERGROUP, Chat.CHANNEL) and self.chat.username:
-            return "https://t.me/{}/{}".format(self.chat.username, self.message_id)
+        """:obj:`str`: Convenience property. If the chat of the message is not
+        a private chat, returns a t.me link of the message."""
+        if self.chat.type != Chat.PRIVATE:
+            if self.chat.username:
+                to_link = self.chat.username
+            else:
+                # this way, we remove the potential minus
+                id_to_link = abs(int(self.chat.id))
```

Comment:
I find this logic confusing.
There isn't a potential minus, there is one. If it's a group, the id is negative.
So I would propose the entire login to change to:
```python
if self.chat.type != Chat.GROUP:
  id_to_link = self.chat.id[3:]
else:
  id_to_link = self.chat.id[1:]
```

Reply:
Done